sensing/position_estimation.cpp

```diff
@@ -356,41 +336,47 @@ Position_estimation::Position_estimation(State& state, Barometer& barometer, con
         sonar(sonar)
 {
     // default GPS home position
-    local_position.origin.longitude =  config.origin.longitude;
-    local_position.origin.latitude =   config.origin.latitude;
-    local_position.origin.altitude =   config.origin.altitude;
+    origin_.longitude =  config.origin.longitude;
+    origin_.latitude =   config.origin.latitude;
+    origin_.altitude =   config.origin.altitude;
 
     for(uint8_t i = 0; i < 3; i++)
     {
-        local_position.pos[i] = 0.0f;
+        local_position[i] = 0.0f;
         kp_pos_gps[i] = config.kp_pos_gps[i];
         kp_vel_gps[i] = config.kp_vel_gps[i];
     }
-
-    if (fence_set)
-    {
-        set_new_fence_origin();
```

Comment:
While we are cleaning, what about moving fence outside 
position_estimation & waypoint_handler & state ? :-)


On 25/07/16 10:01, GregoireH wrote:
>
> In sensing/position_estimation.cpp 
> <https://github.com/lis-epfl/MAVRIC_Library/pull/278#discussion_r72022649>:
>
> >          kp_pos_gps[i] = config.kp_pos_gps[i];
> >          kp_vel_gps[i] = config.kp_vel_gps[i];
> >      }
> > -
> > -    if (fence_set)
> > -    {
> > -        set_new_fence_origin();
>
> TODO,
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/lis-epfl/MAVRIC_Library/pull/278/files/e69a8efd341b2f4d141c92ac525a284073b770cb#r72022649>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABUMh-p1q5r2Jq0qybJjdMHt0POnXgQAks5qZG1sgaJpZM4JOlA9>.
>